python/ray/serve/_private/replica_scheduler/pow_2_scheduler.py

```diff
@@ -442,6 +442,15 @@ async def choose_two_replicas_with_backoff(
                     self.num_scheduling_tasks_in_backoff
                 )
 
+    def drop_replica(self, replica_id: ReplicaID):
+        self._replicas.pop(replica_id, None)
+        self._replica_id_set.discard(replica_id)
+        for id_set in self._colocated_replica_ids.values():
+            id_set.discard(replica_id)
```

Comment:
nit: this is sort of exposing an implementation detail, would prefer to leave the details of what happens to the scheduler implementation:
```suggestion
    def on_replica_actor_died(self, replica_id: ReplicaID):
        """Describe what we do."""
        self._replicas.pop(replica_id, None)
        self._replica_id_set.discard(replica_id)
        for id_set in self._colocated_replica_ids.values():
            id_set.discard(replica_id)
```

Reply:
Ah nice, this is a lot cleaner! renamed to `on_replica_actor_died`